driver/pelican/types/thermSchedule.go

```diff
@@ -0,0 +1,246 @@
+	// Construct Request URL for Thermostat Schedule by Day of Week
+	var requestURL bytes.Buffer
+	requestURL.WriteString(fmt.Sprintf("https://%s.officeclimatecontrol.net/thermDayEdit.cgi?section=json&nodename=", pel.sitename))
+	requestURL.WriteString(thermostatID)
+	requestURL.WriteString("&epnum=")
+	requestURL.WriteString(fmt.Sprintf("%.0f", epnum))
+	requestURL.WriteString("&dayofweek=")
+	requestURL.WriteString(strconv.Itoa(dayOfWeek))
+
+	// Make Request, Decode into Response Struct
+	resp, _, errs := pel.scheduleReq.Get(requestURL.String()).Type("form").AddCookie(pel.cookie).End()
+	if errs != nil {
+		return nil, fmt.Errorf("Failed to retrieve schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, errs)
+	}
+	var result scheduleRequest
+	decoder := json.NewDecoder(resp.Body)
+	if decodeError := decoder.Decode(&result); decodeError != nil {
+		return nil, fmt.Errorf("Failed to decode schedule for thermostat %v on day of week %v: %v", thermostatID, dayOfWeek, decodeError)
+	}
+
+	// Transfer Response Struct Data into return struct
+	var daySchedule []ThermostatBlockSchedule
+	for _, block := range result.ClientData.SetTimes {
+		returnBlock := ThermostatBlockSchedule{
+			CoolSetting: block.CoolSetting,
+			HeatSetting: block.HeatSetting,
+			System:      block.System,
+		}
+
+		if rruleTime, rruleError := convertTimeToRRule(dayOfWeek, block.StartValue, pel.timezone); rruleError != nil {
+			return nil, fmt.Errorf("Failed to convert time in string format %v to rrule format: %v", block.StartValue, rruleError)
+		} else {
+			returnBlock.Time = rruleTime
+		}
+		daySchedule = append(daySchedule, returnBlock)
+	}
+	return &daySchedule, nil
+}
+
+func convertTimeToRRule(dayOfWeek int, blockTime string, timezone *time.Location) (string, error) {
+	timeParsed, timeParsedErr := time.Parse("15:04:PM", blockTime)
+	if timeParsedErr != nil {
+		return "", fmt.Errorf("Error parsing time %v with time.Parse function: %v\n: ", blockTime, timeParsedErr)
+	}
+
+	rruleSched, rruleSchedErr := rrule.NewRRule(rrule.ROption{
+		Freq:    rrule.WEEKLY,
+		Wkst:    weekRRule[dayOfWeek],
+		Dtstart: time.Date(0, 0, 0, timeParsed.Hour(), timeParsed.Minute(), 0, 0, timezone),
+	})
+	if rruleSchedErr != nil {
+		return "", fmt.Errorf("Error creating rruleSchedule object: %v\n", rruleSchedErr)
+	}
+
+	return rruleSched.String(), nil
+}
+
+// Handles setting the Pelican fields (cookie, id) that can only be retrieved by AJAX Requests
+func (pel *Pelican) setCookieAndID() error {
+	// Set the cookie field using the pelican's login information
+	loginInfo := map[string]interface{}{
+		"username": pel.username,
+		"password": pel.password,
+		"sitename": pel.sitename,
+	}
+	respLogin, _, errsLogin := gorequest.New().Post(fmt.Sprintf("https://%s.officeclimatecontrol.net/#_loginPage", pel.sitename)).Type("form").Send(loginInfo).End()
```

Comment:
Any reason we can't just use `pel.scheduleReq` here?

Reply:
No reason! I changed it to scheduleReq. I think that was a remnant of porting this code from pelican.go to this file (since I wasn't sure scheduleReq was instantiated unlike now)

Reply:
I guess from a conceptual point of view I was thinking that these 2 go requests are technically not related to "schedule" retrieval calls, if that makes sense? One is for login and the other is for retrieving an ID.

However, the argument that this is part of the schedule retrieval code + is more efficient with gorequests definitely stands, I agree with using pel.scheduleReq instead!